### PR TITLE
Fixes Docker publish and iroha2-dev push CI

### PR DESCRIPTION
**Pull request by eadventurous**
Signed-off-by: Egor Ivkov <e.o.ivkov@gmail.com>

### Description of the Change
Fixes Docker publish and iroha2-dev push CI

### Benefits
- Docker images will be published on iroha2-dev push
- iroha2-dev CI will not be failing
- Moves coverage and bench into PR CI
- Fixes add label action

### CODE REVIEW

**Review by Revertron — DISMISSED** [review]

**Review by Revertron — APPROVED** [review]

File: .github/workflows/iroha2-dev.yml [hunk 1]
```diff
@@ -100,49 +36,50 @@ jobs:
         uses: actions/upload-artifact@v1
         with:
           name: cargo-build-release
-          path: target/release/iroha
+          path: ./target/release/iroha
```
Comment (i1i1) [inline .github/workflows/iroha2-dev.yml:39]: could you remove `./` and make it `iroha_cli`?

File: .github/workflows/iroha2-dev-pr.yml [hunk 2]
```diff
@@ -239,3 +235,61 @@ jobs:
         run: |
           rm -f ~/.gradle/caches/modules-2/modules-2.lock
           rm -f ~/.gradle/caches/modules-2/gc.properties
+
+  coverage:
```
Comment (i1i1) [inline .github/workflows/iroha2-dev-pr.yml:239]: It is here just for testing, right?

File: .github/workflows/iroha2-dev-pr.yml [hunk 3]
```diff
@@ -121,28 +121,24 @@ jobs:
...
-          sudo apt-get install -y --no-install-recommends \
+          apt-get update
+          apt-get install -y --no-install-recommends \
             build-essential \
             ca-certificates \
             clang \
             llvm-dev \
-            libssl-dev
+            libssl-dev \
+            docker-compose
```
[the remark this answers is not in the document]
Comment (i1i1) [reply .github/workflows/iroha2-dev-pr.yml:141]: (related to docker-compose)

File: .github/workflows/iroha2-dev-pr.yml (hunk 2 above)
Comment (eadventurous) [reply .github/workflows/iroha2-dev-pr.yml:243]: About coverage I am checking how the codecov bot works, it's said it generates nice reports for PRs, if so then I think we can have it in our PR workflow

Comment (eadventurous) [reply .github/workflows/iroha2-dev-pr.yml:243]: Actually I am thinking that for our push workflow only archive and publish should be there

**Review by i1i1 — APPROVED** [review]

**Pull request merged**